File: app/src/main/java/com/example/boulderjournal/data/RouteDao.java
package com.example.boulderjournal.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface RouteDao {

    @Query("SELECT * FROM route")
    LiveData<List<RouteEntry>> loadAllRoutes();

    @Insert
    void insertRoute(RouteEntry routeEntry);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateRoute(RouteEntry routeEntry);

    @Delete
    void deleteRoute(RouteEntry routeEntry);

    @Query("SELECT * FROM route WHERE id = :id")
    RouteEntry loadRouteById(int id);

    @Query("SELECT * FROM route WHERE complete = 'false'")
    LiveData<List<RouteEntry>> loadUnfinishedRoutes();

    @Query("SELECT * FROM route WHERE complete = 'true'")
    LiveData<List<RouteEntry>> loadFinishedRoutes();

}

File: app/src/main/java/com/example/boulderjournal/data/RouteDao.kt
package com.example.boulderjournal.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

@Dao
interface RouteDao {

    @Query("SELECT * FROM route")
    fun loadAllRoutes(): LiveData<List<RouteEntry>>?

    @Insert
    fun insertRoute(routeEntry: RouteEntry)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateRoute(routeEntry: RouteEntry)

    @Delete
    fun deleteRoute(routeEntry: RouteEntry)

    @Query("SELECT * FROM route WHERE id = :id")
    fun loadRouteById(id: Int): RouteEntry

    @Query("SELECT * FROM route WHERE complete = 'false'")
    fun loadUnfinishedRoutes(): LiveData<List<RouteEntry>>

    @Query("SELECT * FROM route WHERE complete = 'true'")
    fun loadFinishedRoutes(): LiveData<List<RouteEntry>>?

}
